src/dev/i18n/utils.js

```diff
@@ -84,3 +94,71 @@ export function* traverseNodes(nodes) {
     }
   }
 }
+
+/**
+ * @param {string[]} valuesKeys array of "values" property keys
+ * @param {string} defaultMessage "defaultMessage" value
+ * @throws if "values" and "defaultMessage" don't correspond to each other
+ */
+export function checkValuesProperty(valuesKeys, defaultMessage, messageId) {
+  const defaultMessageReferences = defaultMessage.match(VALUES_REFERENCES_REGEX);
```

Comment:
__issue:__ hmm, I do think we should use `intl-messageformat-parser` instead of our own regex whenever we want to parse default  message. This way we will use the same ICU parser that our engine uses. Is there any reason why you don't want to use it?

Reply:
No, we just didn't take this parser into account. 
I fixed this.